testkit-backend/src/main/java/neo4j/org/testkit/backend/Runner.java

```diff
@@ -16,41 +16,59 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 import java.io.BufferedReader;
 import java.io.BufferedWriter;
-import java.io.UncheckedIOException;
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.OutputStreamWriter;
+import java.io.UncheckedIOException;
 import java.net.ServerSocket;
 import java.net.Socket;
+import java.util.concurrent.CompletableFuture;
 
 public class Runner
 {
     public static void main( String[] args ) throws IOException
     {
-        ServerSocket serverSocket = new ServerSocket(9876);
-        System.out.println("Starting Java Testkit Backend Started");
+        ServerSocket serverSocket = new ServerSocket( 9876 );
```

Comment:
I would strongly recommend using the local loop back address instead of the wildcard (this is what the standard ctor does, using 0.0.0.0). I have had issues with several projects on a couple of machines.

Reply:
Local loop back will not work when the backend is running in a container (which it normally is)

Reply:
Just to be sure, I tested and It didn't work.

Reply:
Ok, makes sense @2hdddg 

The clients connecting against the backend: Are they on the same network docker network? Or coming from the external? If so,  shouldn't it than use the containers actual network address? 

But alas, if it works, it's all good. I just had this issue with the maven daemon like 3 days ago and therefore I noticed it.

Reply:
It will work with 0.0.0.0, can't see that it should be a problem in docker, could be when running on host. If you want you can add a commandline param to control the interface to bind to and bind to localhost when running outside of docker.
The clients are on the same network (dynamically created), we connect through host names so we would need to figure out (or configure) the ip address of the driver container if we for some reason think we must bind to that exact interface.
It would work to bind to localhost if we use host network instead of a bridge but that is like asking for problems.